pandas/core/window/expanding.py

```diff
@@ -170,92 +193,275 @@ def apply(
             kwargs=kwargs,
         )
 
-    @Substitution(name="expanding")
-    @Appender(_shared_docs["sum"])
+    @doc(
+        doc_template,
+        window_method="expanding",
+        aggregation_description="sum",
+        parameters=window_agg_numba_args_kwargs_parameters,
+        numpy_args_kwargs="",
+        agg_method="sum",
+        other_see_also="",
+        notes=numba_notes,
+        examples="",
+    )
     def sum(self, *args, engine=None, engine_kwargs=None, **kwargs):
         nv.validate_expanding_func("sum", args, kwargs)
         return super().sum(*args, engine=engine, engine_kwargs=engine_kwargs, **kwargs)
 
-    @Substitution(name="expanding", func_name="max")
-    @Appender(_doc_template)
-    @Appender(_shared_docs["max"])
+    @doc(
+        doc_template,
+        window_method="expanding",
+        aggregation_description="maximum",
+        parameters=window_agg_numba_args_kwargs_parameters,
+        numpy_args_kwargs="",
+        agg_method="max",
+        other_see_also="",
+        notes=numba_notes,
+        examples="",
+    )
     def max(self, *args, engine=None, engine_kwargs=None, **kwargs):
         nv.validate_expanding_func("max", args, kwargs)
         return super().max(*args, engine=engine, engine_kwargs=engine_kwargs, **kwargs)
 
-    @Substitution(name="expanding")
-    @Appender(_shared_docs["min"])
+    @doc(
+        doc_template,
+        window_method="expanding",
+        aggregation_description="minimum",
+        parameters=window_agg_numba_args_kwargs_parameters,
+        numpy_args_kwargs="",
+        agg_method="min",
+        other_see_also="",
+        notes=numba_notes,
+        examples="",
+    )
     def min(self, *args, engine=None, engine_kwargs=None, **kwargs):
         nv.validate_expanding_func("min", args, kwargs)
         return super().min(*args, engine=engine, engine_kwargs=engine_kwargs, **kwargs)
 
-    @Substitution(name="expanding")
-    @Appender(_shared_docs["mean"])
+    @doc(
+        doc_template,
+        window_method="expanding",
+        aggregation_description="mean",
+        parameters=window_agg_numba_args_kwargs_parameters,
+        numpy_args_kwargs="",
+        agg_method="mean",
+        other_see_also="",
+        notes=numba_notes,
+        examples="",
+    )
     def mean(self, *args, engine=None, engine_kwargs=None, **kwargs):
         nv.validate_expanding_func("mean", args, kwargs)
         return super().mean(*args, engine=engine, engine_kwargs=engine_kwargs, **kwargs)
 
-    @Substitution(name="expanding")
-    @Appender(_shared_docs["median"])
+    @doc(
+        doc_template,
+        window_method="expanding",
+        aggregation_description="median",
+        parameters=window_agg_numba_parameters,
```

Comment:
**kwargs not in parameters. (OT for a doc PR, median is inconsistent with mean, missing *args)